interceptors/logging/interceptors_test.go

```diff
@@ -501,3 +537,271 @@ func (s *loggingCustomDeciderSuite) TestPingList_HasCustomDecider() {
 
 	require.Len(s.T(), s.logger.Lines(), 0) // Decider should suppress.
 }
+
+type requestLoggingSuite struct {
+	*baseLoggingSuite
+}
+
+func TestRequestLoggingSuite(t *testing.T) {
+
+	if strings.HasPrefix(runtime.Version(), "go1.7") {
+		t.Skip("Skipping due to json.RawMessage incompatibility with go1.7")
+		return
+	}
+	// Switch on the request logging.
+	opts := logging.WithDecider(func(_ string, _ error) bool {
+		return true
+	})
+
+	s := &requestLoggingSuite{
+		baseLoggingSuite: &baseLoggingSuite{
+			logger: &mockLogger{baseMockLogger: &baseMockLogger{}},
+			InterceptorTestSuite: &grpctesting.InterceptorTestSuite{
+				TestService: &grpctesting.TestPingService{T: t},
+			},
+		},
+	}
+
+	s.InterceptorTestSuite.ClientOpts = []grpc.DialOption{
+		grpc.WithUnaryInterceptor(logging.UnaryClientInterceptor(s.logger, opts)),
+		grpc.WithStreamInterceptor(logging.StreamClientInterceptor(s.logger, opts)),
+	}
+	s.InterceptorTestSuite.ServerOpts = []grpc.ServerOption{
+		middleware.WithStreamServerChain(
+			tags.StreamServerInterceptor(tags.WithFieldExtractor(tags.CodeGenRequestFieldExtractor)),
+			logging.StreamServerInterceptor(s.logger, opts)),
+		middleware.WithUnaryServerChain(
+			tags.UnaryServerInterceptor(tags.WithFieldExtractor(tags.CodeGenRequestFieldExtractor)),
+			logging.UnaryServerInterceptor(s.logger, opts)),
+	}
+	suite.Run(t, s)
+}
+
+func (s *requestLoggingSuite) TestPing_RequestLogging() {
```

Comment:
Do we need to duplicae all of this code? Sounds like we would rather extend the normal reporter test, as we kind of reuse the same thing.

Reply:
Yup, extending the tests are better